### ISSUE TITLE
Can't parse object

### ISSUE THREAD

**Issue by kopyl** [dpranke/pyjson5#70]
```
import json5


results_json = """{
    title: "\"Picnic collective\" logo and characters",
}"""

json5.loads(results_json)
```

And you get this error:
```
ValueError: <string>:2 Unexpected "P" at column 14
```

Could you please recommend any other libraries which can JUST WORK in ALL possible cases?
I just don't want my code to break if i get some unexpected JS object.

**Comment by dpranke** [comment]
I think this isn't a bug in my library. I believe Python translates an escaped double quote into a regular double quote. In other words, "\"" == '"'.

So, your results json ends up reading as '{title: ""Picnic collective" logo and characters}'.

Which parses as '{title: ""' and then the illegal character 'P', the 'P' being illegal since it follows the double-quoted empty string.

This is similar to #68, but #68 is in fact a bug in the parser.

At least, that's my current thinking. I'm definitely open to discussing this in case I'm missing something.

> Could you please recommend any other libraries which can JUST WORK in ALL possible cases?

#68 is the only bug I know of in the library where we don't parse something legal. I haven't gotten around to fixing it mostly because I've been lazy, but also because I didn't think it was really affecting anyone in the real world. If you are being held back by it, please let me know and I'll try to prioritize fixing it.

**Comment by kopyl** [comment]
So fixing the #68 will fix this?

No need to rush, I switched to some other project for this time...

**Comment by dpranke** [comment]
> So fixing the https://github.com/dpranke/pyjson5/issues/68 will fix this?

No. As I tried to explain above, I don't believe the issue you're hitting is a bug to be fixed. I think the code is doing the right thing.

**Comment by dpranke** [comment]
> > So fixing the https://github.com/dpranke/pyjson5/issues/68 will fix this?

> No. As I tried to explain above, I don't believe the issue you're hitting is a bug to be fixed. I think the code is doing the right thing.

And, since #68 is talking about the same problem, I've gone back and reviewed that bug as well and I think that's also not a bug.

**Comment by kopyl** [comment]
If you don't call it a bug, then how can I parse any JS dictionary with your library?

**Comment by dpranke** [comment]
I apologize, I don't think I'm explaining myself well. Let me try again:

```
$ python3
Python 3.10.9 (main, Jan 23 2023, 19:36:15) [Clang 14.0.0 (clang-1400.0.29.102)] on darwin
Type "help", "copyright", "credits" or "license" for more information.
>>> import json5
>>> results_json = """{
...     title: "\"Picnic collective\" logo and characters",
... }"""
>>> results_json2 = = """{
...     title: ""Picnic collective" logo and characters",
... }"""
>>> results_json == results_json2
True
>>> results_json3 = r"""{
...     title: "\"Picnic collective\" logo and characters",
... }"""
>>> json5.loads(results_json3)
{'title': '"Picnic collective" logo and characters'}
>>>
$ 
```

Hopefully the example above makes it clear that Python treats those first two strings as if they are identical, because of the way Python handles escape characters in strings, and that the second string isn't legal JSON5. If you use an r-string, instead, the escape char gets passed along uninterpreted and thus the resulting thing becomes legal.

Does that help?

**Comment by dpranke** [comment]
Closing this since there's been no reply.
